torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -403,6 +403,23 @@ const std::vector<std::string> functions = {
                 return grad_self, grad_other
 
             return torch.matmul(self, other), backward
+
+        def linear(input : Tensor,
+                   weight : Tensor,
+                   bias : Optional[Tensor]):
+            result = torch.linear(input, weight, bias)
```

Comment:
what's torch.linear and how does this work?

Reply:
This goes through:
https://github.com/pytorch/pytorch/blob/00675292ca52aa459690b9a31f5893f53dcf8659/aten/src/ATen/native/native_functions.yaml#L2133-L2135
https://github.com/pytorch/pytorch/blob/00675292ca52aa459690b9a31f5893f53dcf8659/aten/src/ATen/native/Linear.cpp#L30-L48

Reply:
Nope, because native_functions entry has `python_module: nn`, it's not available as torch.linear (try it), it's only available as torch._C._nn.linear (that's what `linear` in torch/nn/functiona.py calls).

Reply:
strings in symbolic_script.cpp is parsed directly, so it has nothing to do with how aten is exposed in python.
specifically here, we are using compilation unit to parse the string directly and `torch` is resolved by NativeResolver where they map it to `aten`, so it is mapped to `aten::linear`: https://github.com/pytorch/pytorch/blob/fc1153a8be7cb1d4e7d1219746dc744cb8e28b4f/torch/csrc/jit/frontend/resolver.h#L48-L56

Reply:
At least that's how I understand this... Granted that I know very little about jit frontend so I'm cc'ing @Krovatkin here just in case I'm embarrassing myself again :crossed_fingers:

Reply:
THis is super confusing, but plausible